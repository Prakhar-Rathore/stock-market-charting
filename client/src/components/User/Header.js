import React from 'react';
import MuiAppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import { Grid, Tooltip, Menu, Button, List, ListItemText, Avatar, TextField, Autocomplete } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import SearchIcon from '@mui/icons-material/Search';
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import MenuItem from '@mui/material/MenuItem';
import Typography from '@mui/material/Typography';
import { styled, alpha, useTheme } from '@mui/material/styles';
import InputBase from '@mui/material/InputBase';
import { useAuth } from "../../context/AuthContext";
import { auth } from "../../firebase";
import { Navigate, Link } from 'react-router-dom';
import Paper from '@mui/material/Paper';
import MuiDrawer from "@mui/material/Drawer";
import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";
import ChevronRightIcon from "@mui/icons-material/ChevronRight";
import Divider from "@mui/material/Divider";
import BusinessIcon from "@mui/icons-material/Business";
import CompareIcon from "@mui/icons-material/Compare";
import ListItem from "@mui/material/ListItem";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemIcon from "@mui/material/ListItemIcon";
import { routes } from "../../config/serverconfig";
import apiCall from "../../apiCall/apiCall";
import AdminPanelSettingsIcon from '@mui/icons-material/AdminPanelSettings';
import DifferenceIcon from '@mui/icons-material/Difference';

const PROD_SERV_ADDRESS_API = routes.PROD_SERV_ADDRESS_API;

const drawerWidth = 240;

const openedMixin = (theme) => ({
    width: drawerWidth,
    transition: theme.transitions.create("width", {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen,
    }),
    overflowX: "hidden",
});

const closedMixin = (theme) => ({
    transition: theme.transitions.create("width", {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
    }),
    overflowX: "hidden",
    width: `calc(${theme.spacing(7)} + 1px)`,
    [theme.breakpoints.up("sm")]: {
        width: `calc(${theme.spacing(8)} + 1px)`,
    },
});

const DrawerHeader = styled("div")(({ theme }) => ({
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
}));

const Drawer = styled(MuiDrawer, {
    shouldForwardProp: (prop) => prop !== "open",
})(({ theme, open }) => ({
    width: drawerWidth,
    flexShrink: 0,
    whiteSpace: "nowrap",
    boxSizing: "border-box",
    ...(open && {
        ...openedMixin(theme),
        "& .MuiDrawer-paper": openedMixin(theme),
    }),
    ...(!open && {
        ...closedMixin(theme),
        "& .MuiDrawer-paper": closedMixin(theme),
    }),
    background: "radial-gradient(90% 100% at 50% 0%, #2d685a 0%, #050606 100%), radial-gradient(90% 100% at 50% 0%, #a86487 0%, #827878 100%), radial-gradient(110% 215% at 100% 0%, #000AFF 0%, #3A5525 100%), linear-gradient(230deg, #000 0%, #09FF04 100%), linear-gradient(130deg, #2D2929 0%, #C4B6FB 100%)",
    backgroundBlendMode: "overlay, difference, difference, exclusion, normal"
}));

const AppBar = styled(MuiAppBar, {
    shouldForwardProp: (prop) => prop !== 'open',
})(({ theme, open }) => ({
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
    }),
    ...(open && {
        marginLeft: drawerWidth,
        width: `calc(100% - ${drawerWidth}px)`,
        transition: theme.transitions.create(['width', 'margin'], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.enteringScreen,
        }),
    }),
}));


export default function Header({ setCompanyData }) {

    const theme = useTheme();
    const [open, setOpen] = React.useState(false);

    const { logout } = useAuth();

    const styles = theme => ({
        icon: {
            fill: 'white'
        }
    })

    // const [company, setCompany] = React.useState("");
    const [anchorElUser, setAnchorElUser] = React.useState(null);
    const [idToken, setIdToken] = React.useState(null);
    const [isAdmin, setIsAdmin] = React.useState(false);

    const classes = styles;

    const handleOpenUserMenu = (event) => {
        setAnchorElUser(event.currentTarget);
    };

    const handleCloseUserMenu = () => {
        setAnchorElUser(null);
    };

    const handleLogout = async (event) => {
        event.preventDefault();
        try {
            var res = await logout();
            console.log(res);
        } catch (error) {
            console.log(error);
        }
    }

    const handleDrawerOpen = () => {
        setOpen(true);
    };

    const handleDrawerClose = () => {
        setOpen(false);
    };

    React.useEffect(() => {
        (async () => {
            const token = await auth.currentUser.getIdToken();
            setIdToken(token);
            const response = await apiCall(
                `${PROD_SERV_ADDRESS_API}/admin/isAdmin`,
                "GET",
                null,
                token
            );
            if (response.data.message === 'Admin') setIsAdmin(true)
        })();
    }, []);

    if (!auth.currentUser) {
        return <Navigate to="/" />;
    }
    else {
        return (
            <Box>
                <AppBar
                    position="fixed"
                    open={open}
                    sx={{
                        background: "radial-gradient(90% 100% at 50% 0%, #2d685a 0%, #050606 100%), radial-gradient(90% 100% at 50% 0%, #a86487 0%, #827878 100%), radial-gradient(110% 215% at 100% 0%, #000AFF 0%, #3A5525 100%), linear-gradient(230deg, #000 0%, #09FF04 100%), linear-gradient(130deg, #2D2929 0%, #C4B6FB 100%)",
                        backgroundBlendMode: "overlay, difference, difference, exclusion, normal",
                        height: "64px"
                    }}
                >
                    <Toolbar style={{ paddingRight: 0 }}>
                        <Grid container>
                            <Grid item xs={1} display="flex">
                                <IconButton
                                    edge="start"
                                    color="inherit"
                                    aria-label="open drawer"
                                    onClick={handleDrawerOpen}
                                    sx={{
                                        ...(open && { display: "none" })
                                    }}
                                >
                                    <MenuIcon style={{ alignSelf: "center" }} />
                                </IconButton>
                                <Box sx={{ display: { xs: 'none', sm: 'block' } }}>
                                    <Link to="/user-home" onClick={() => setCompanyData("")}>
                                        <IconButton>
                                            <Avatar
                                                src="https://cdn2.vectorstock.com/i/1000x1000/67/86/mobile-stock-market-business-logo-icon-design-vector-22926786.jpg"
                                                alt="logo"
                                                sx={{ ...(open && { display: "none" }) }}
                                            />
                                        </IconButton>
                                    </Link>
                                </Box>
                            </Grid>
                            <Grid item display="flex" flex={1} xs={10} alignItems="center" fontFamily="Segoe Script" justifyContent="center">
                                <div>Stock Market Charting</div>
                            </Grid>
                            <Grid item xs={1} style={{ display: "flex", justifyContent: "flex-end" }}>
                                <Tooltip title="Open profile options">
                                    <IconButton
                                        size="large"
                                        color="inherit"
                                        aria-label="open profile options"
                                        sx={{ mr: 1, height: "100%" }}
                                        onClick={handleOpenUserMenu}
                                    >
                                        <AccountCircleIcon />
                                    </IconButton>
                                </Tooltip>
                                <Menu
                                    sx={{ mt: '45px' }}
                                    id="menu-appbar"
                                    anchorEl={anchorElUser}
                                    anchorOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    keepMounted
                                    transformOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    open={Boolean(anchorElUser)}
                                    onClose={handleCloseUserMenu}
                                >
                                    <Link to="/reset-pass" style={{ color: "inherit", textDecoration: "none" }}>
                                        <MenuItem key="forgotPass">
                                            Change Password
                                        </MenuItem>
                                    </Link>
                                    <MenuItem key="Logout" onClick={handleLogout}>
                                        <Button sx={{ color: "inherit", textTransform: "inherit", fontSize: '1rem', padding: '0' }}>Logout</Button>
                                    </MenuItem>
                                </Menu>
                            </Grid>
                        </Grid>
                    </Toolbar>
                </AppBar>
                <Drawer variant="permanent" open={open} PaperProps={{
                    sx: {
                        background: "radial-gradient(90% 100% at 50% 0%, #2d685a 0%, #050606 100%), radial-gradient(90% 100% at 50% 0%, #a86487 0%, #827878 100%), radial-gradient(110% 215% at 100% 0%, #000AFF 0%, #3A5525 100%), linear-gradient(230deg, #000 0%, #09FF04 100%), linear-gradient(130deg, #2D2929 0%, #C4B6FB 100%)",
                        backgroundBlendMode: "overlay, difference, difference, exclusion, normal",
                        color: "white"
                    }
                }}>
                    <DrawerHeader>
                        <Link to="/user-home" style={{ position: "absolute", left: "10px" }}>
                            <IconButton>
                                <Avatar
                                    src="https://cdn2.vectorstock.com/i/1000x1000/67/86/mobile-stock-market-business-logo-icon-design-vector-22926786.jpg"
                                    alt="logo"
                                    sx={{ width: 56, height: 56 }}
                                />
                            </IconButton>
                        </Link>
                        <IconButton onClick={handleDrawerClose}>
                            {theme.direction === 'rtl' ? <ChevronRightIcon style={{ color: "white" }} /> : <ChevronLeftIcon style={{ color: "white" }} />}
                        </IconButton>
                    </DrawerHeader>
                    <Divider />
                    <List>
                        <ListItem key="search company" disablePadding sx={{ display: "block" }}>
                            <Link to="/user-home" style={{ textDecoration: "none", color: "white" }}>
                                <ListItemButton
                                    sx={{
                                        minHeight: 48,
                                        justifyContent: open ? "initial" : "center",
                                        px: 2.5,
                                    }}
                                >
                                    <Tooltip title="Search for a company">
                                    <ListItemIcon
                                        sx={{
                                            minWidth: 0,
                                            mr: open ? 3 : "auto",
                                            justifyContent: "center",
                                            color: "white"
                                        }}
                                    >
                                        <SearchIcon />
                                    </ListItemIcon>
                                     </Tooltip>
                                    <ListItemText
                                        primary="Search company"
                                        sx={{ opacity: open ? 1 : 0 }}
                                    />
                                </ListItemButton>
                            </Link>
                        </ListItem>
                        <ListItem key="IPO Dashboard" disablePadding sx={{ display: "block" }}>
                            <Link to="/ipo-dashboard" style={{ textDecoration: "none", color: "white" }}>
                                <ListItemButton
                                    sx={{
                                        minHeight: 48,
                                        justifyContent: open ? "initial" : "center",
                                        px: 2.5,
                                    }}
                                >
                                    <Tooltip title="IPO dashboard">
                                    <ListItemIcon
                                        sx={{
                                            minWidth: 0,
                                            mr: open ? 3 : "auto",
                                            justifyContent: "center",
                                            color: "white"
                                        }}
                                    >
                                        <BusinessIcon />
                                    </ListItemIcon>
                                    </Tooltip>
                                    <ListItemText
                                        primary="IPO Dashboard"
                                        sx={{ opacity: open ? 1 : 0 }}
                                    />
                                </ListItemButton>
                            </Link>
                        </ListItem>
                        <ListItem key="Compare" disablePadding sx={{ display: "block" }}>
                            <Link to="/compare" style={{ textDecoration: "none", color: "white" }}>
                                <ListItemButton
                                    sx={{
                                        minHeight: 48,
                                        justifyContent: open ? "initial" : "center",
                                        px: 2.5,
                                    }}
                                >
                                       <Tooltip title="Company comparison">
                                    <ListItemIcon
                                        sx={{
                                            minWidth: 0,
                                            mr: open ? 3 : "auto",
                                            justifyContent: "center",
                                            color: "white"
                                        }}
                                    >
                                        <CompareIcon />
                                    </ListItemIcon>
                                    </Tooltip>
                                    <ListItemText
                                        primary="Compare"
                                        sx={{ opacity: open ? 1 : 0 }}
                                    />
                                </ListItemButton>
                            </Link>
                        </ListItem>
                        <ListItem key="Sector Scan" disablePadding sx={{ display: "block" }}>
                            <Link to="/sector-scan" style={{ textDecoration: "none", color: "white" }}>
                                <ListItemButton
                                    sx={{
                                        minHeight: 48,
                                        justifyContent: open ? "initial" : "center",
                                        px: 2.5,
                                    }}
                                >
                                       <Tooltip title="Sector scan">
                                    <ListItemIcon
                                        sx={{
                                            minWidth: 0,
                                            mr: open ? 3 : "auto",
                                            justifyContent: "center",
                                            color: "white"
                                        }}
                                    >
                                        <DifferenceIcon />
                                    </ListItemIcon>
                                    </Tooltip>
                                    <ListItemText
                                        primary="Sector Scan"
                                        sx={{ opacity: open ? 1 : 0 }}
                                    />
                                </ListItemButton>
                            </Link>
                        </ListItem>
                        {isAdmin && <ListItem key="Admin Dashboard" disablePadding sx={{ display: "block" }}>
                            <Link to="/manage-users" style={{ textDecoration: "none", color: "white" }}>
                                <ListItemButton
                                    sx={{
                                        minHeight: 48,
                                        justifyContent: open ? "initial" : "center",
                                        px: 2.5,
                                    }}
                                >
                                       <Tooltip title="Admin dashboard">
                                    <ListItemIcon
                                        sx={{
                                            minWidth: 0,
                                            mr: open ? 3 : "auto",
                                            justifyContent: "center",
                                            color: "white"
                                        }}
                                    >
                                        <AdminPanelSettingsIcon />
                                    </ListItemIcon>
                                    </Tooltip>
                                    <ListItemText
                                        primary="Admin Dashboard"
                                        sx={{ opacity: open ? 1 : 0 }}
                                    />
                                </ListItemButton>
                            </Link>
                        </ListItem>}
                    </List>
                    <Divider />
                </Drawer>
            </Box>
        );
    }
}