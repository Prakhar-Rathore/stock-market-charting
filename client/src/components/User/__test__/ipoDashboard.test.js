import React from "react";
import ReactDOM from "react-dom/client";
import IpoDashboard from "./../ipoDashboard";
import Header from "./../Header";
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';

jest.mock("./../Header", () => {
    return {
      __esModule: true,
      default: () => {
        return <div/>;
      },
      Header: () => {
        return <div/>;
      },
    };
  });

it("renders without crashing", async() => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act( async () => render(<IpoDashboard/>));
});

it("renders correctly", () => {
  render(<IpoDashboard/>)
})

it("matches snapshot", () => {
  const tree = renderer.create(<IpoDashboard/>).toJSON();
  expect(tree).toMatchSnapshot();
})
