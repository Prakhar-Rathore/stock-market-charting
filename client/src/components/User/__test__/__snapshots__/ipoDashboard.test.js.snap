// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <main>
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Open modal
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </main>,
  <div />,
  <div
    className="MuiBox-root css-1cako5s"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-y6l5n9-MuiGrid-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        style={
          Object {
            "coSelf": "center",
            "margin": "5px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <div
        aria-owns={null}
        className="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1auycx3-MuiAutocomplete-root"
        color="primary"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "border": "1px solid white",
            "borderRadius": "15px",
            "padding": "10px",
            "width": "75%",
          }
        }
      >
        <div
          className="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-7hojm3-MuiInputBase-root"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-expanded={false}
            autoCapitalize="none"
            autoComplete="off"
            className="MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-ynu1c4-MuiInputBase-input"
            disabled={false}
            id="input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            placeholder="Search for a company..."
            readOnly={false}
            role="combobox"
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-1ciwbrp-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="toggle between light and dark mode"
      aria-labelledby={null}
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-gjqkap-MuiButtonBase-root-MuiButton-root"
      data-mui-internal-clone-element={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DarkModeIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-grid-xs-12 css-9fc0yf-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-4881pg-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root css-vtx92c-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-wxr07q-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          id="SE"
        >
          IPO MainBoard
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-533s6m-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          onClick={[Function]}
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="SE SE"
            className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-v3zyv7-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="SE"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1wu7ecg-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                SE
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-grid-xs-6 css-1m5fj3w-MuiGrid-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body3 css-2imcwd-MuiTypography-root"
        style={
          Object {
            "marginTop": "5px",
          }
        }
      >
        IPO DASHBOARD
      </span>
    </div>
  </div>,
  <div
    className="MuiBox-root css-fo6wgm"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-spacing-md-2 MuiGrid-spacing-lg-2 css-1ffx5kp-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-grid-xs-12 css-ycb5f5-MuiGrid-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-1exk4wk-MuiPaper-root-MuiTableContainer-root"
        >
          <table
            className="MuiTable-root MuiTable-stickyHeader css-1ut88b9-MuiTable-root"
            role={null}
          >
            <thead
              className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head css-1pgphf0-MuiTableRow-root"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-65td30-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                      "textAlign": "center",
                      "width": "20%",
                    }
                  }
                >
                  Company
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-xs7j60-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Exchange
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-xs7j60-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Status
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-xs7j60-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  SecurityType
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-xs7j60-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Symbol
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-xs7j60-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                      "width": "40%",
                    }
                  }
                >
                  Open
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-xs7j60-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                      "width": "40%",
                    }
                  }
                >
                  Close
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-xs7j60-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  IssuePrice(â‚¹)
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-xs7j60-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  IssueSize(Rs.Cr)
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-xs7j60-MuiTableCell-root"
                  scope="header"
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  MarketLot
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root css-1sknl19-MuiTableBody-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root css-1pgphf0-MuiTableRow-root"
                role={null}
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1mx464q-MuiTableCell-root"
                  scope="row"
                >
                  <span
                    className="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
                  >
                    No Data to Display
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "padding": "10px",
      }
    }
  />,
]
`;
